services/user.py

```diff
@@ -0,0 +1,47 @@
+from db.models import User
+
+
+def create_user(
+        username: str,
+        password: str,
+        email: str = None,
+        first_name: str = None,
+        last_name: str = None
+) -> None:
+    user = User(username=username)
+    user.set_password(password)
+    if email:
+        user.email = email
+    if password:
+        user.set_password(password)
```

Comment:
The call to `set_password` is redundant here since it is already called on line 12. You can remove this line to avoid unnecessary operations.